app/src/organisms/RunDetails/ProtocolSetupInfo.tsx

```diff
@@ -0,0 +1,172 @@
+  onCloseClick: () => unknown
+  SetupCommand?: Command
+  runStatus: string
+  type: Status
+}
+
+export const ProtocolSetupInfo = (
+  props: ProtocolSetupInfoProps
+): JSX.Element | null => {
+  const { SetupCommand, runStatus, type } = props
+  const { t } = useTranslation('run_details')
+  const protocolData: ProtocolFile<{}> | null = useProtocolDetails()
+    .protocolData
+  const robot = useSelector((state: State) => getConnectedRobot(state))
+  const robotName = robot?.name != null ? robot?.name : ''
+  const protocolPipetteData = useSelector((state: State) =>
+    getProtocolPipetteTipRackCalInfo(state, robotName)
+  )
+  if (protocolData == null) return null
+  if (SetupCommand === undefined) return null
+
+  let SetupCommandText
+  if (SetupCommand.commandType === 'loadPipette') {
+    const pipetteData = protocolPipetteData[SetupCommand.params.mount]
+    if (pipetteData == null) {
+      return null
+    }
+    SetupCommandText = (
+      <Trans
+        t={t}
+        id={`RunDetails_PipetteSetup`}
+        i18nKey={'load_pipette_protocol_setup'}
+        values={{
+          pipette_name: pipetteData.pipetteDisplayName,
+          mount_name: SetupCommand.params.mount,
+        }}
+        components={{
+          span: (
+            <Text
+              textTransform={TEXT_TRANSFORM_CAPITALIZE}
+              marginLeft={SPACING_1}
+              marginRight={SPACING_1}
+            />
+          ),
+        }}
+      />
+    )
+  } else if (SetupCommand.commandType === 'loadModule') {
+    const moduleId = SetupCommand.params.moduleId
+    const moduleModel = protocolData.modules[moduleId]
+    const moduleSlotNumber = moduleId.includes('thermocycler') ? 4 : 1
+    SetupCommandText = (
+      <Trans
+        t={t}
+        id={`RunDetails_ModuleSetup`}
+        i18nKey={'load_modules_protocol_setup'}
+        count={moduleSlotNumber}
+        values={{
+          module: getModuleDisplayName(moduleModel.model),
+          slot_name: Object.values(SetupCommand.params.location)[0],
+        }}
+      />
+    )
+  } else if (SetupCommand.commandType === 'loadLabware') {
+    let moduleName: string | null = null
+    let slotNumber = Object.values(SetupCommand.params.location)[0]
+    if ('moduleId' in SetupCommand.params.location) {
+      const moduleId = SetupCommand.params.location.moduleId
+      const moduleModel = protocolData.modules[moduleId]
+      const moduleSlotNumber = protocolData.commands.find(
+        command =>
+          command.commandType === 'loadModule' &&
+          command.params.moduleId === moduleId
+        //  @ts-expect-error narrow to load module command when command types get updated
+      )?.params.location.slotName
+      slotNumber = moduleSlotNumber
+      moduleName = getModuleDisplayName(moduleModel.model)
+    }
+    let moduleSlots = 0
+    if (moduleName === null) {
+      moduleSlots = 0
+    } else if (moduleName != null && moduleName.includes('Thermocycler')) {
```

Comment:
this can be replaced by
```suggestion
    } else if (moduleName?.includes('Thermocycler')) {
```